py-rse/py-scripting.Rmd

```diff
@@ -767,7 +918,70 @@ by writing Python scripts that can be executed at the command line.
 
 ## Exercises {#py-rse-py-scripting-exercises}
 
-TODO
+### Running Python statements from the command line
+
+We don't need to open the interactive interpreter to run Python code.
+Instead,
+we can invoke Python with the command flag `-c`
+and the statement we want to run:
+
+```python
+$ python -c "print(2+3)"
+```
+
+```text
+5
+```
+
+When and why is this useful?
+
+### Collating while counting
+
+FIXME modify the counting program to handle multiple files and discuss tradeoffs
+
+### Optional output file
+
+FIXME allow an optional output file argument
+
+### Multiple files and standard input
+
+We have used `nargs='*'` to get `collate.py` to read from zero or more files:
+
+```python
+parser.add_argument('infiles', type=str, nargs='*', help='Input file names')
+```
+
+However,
+if no filenames are specified,
+it does *not* read from standard input.
+Modify the argument handling and the `main` function so that it can do this.
+
+### Ignoring `__pycache__`
+
+Why don't we want to put the byte code files in `__pycache__` in version control?
+
+### A better plotting program
```

Comment:
I wonder how we should highlight that this exercise is mandatory? (i.e. the `plotcounts.py` script is required in subsequent chapters)

Reply:
Also, the solution will need to go in `solutions.Rmd` so that people have the same version of `plotcounts.py` going forward (because further edits to the script are made in subsequent chapters).

Reply:
Added a note in the exercise, left a FIXME in `solutions.Rmd`.